crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -196,6 +167,75 @@ impl SyncService {
         .instrument(tracing::span!(Level::WARN, "supervisor task"))
     }
 
+    async fn shutdown(self) -> Result<(), Error> {
+        match self
+            .abortion_sender
+            .send(TerminationReport {
+                is_error: false,
+                has_expired: false,
+                origin: TerminationOrigin::Supervisor,
+            })
+            .await
+        {
+            Ok(_) => self.task.await.map_err(|err| {
+                error!("couldn't finish supervisor task: {err}");
+                Error::InternalSupervisorError
+            }),
+            Err(err) => {
+                error!("when sending termination report: {err}");
+                self.task.abort();
+                Err(Error::InternalSupervisorError)
+            }
+        }
+    }
+}
+
+struct SyncServiceInner {
+    room_list_service: Arc<RoomListService>,
+    encryption_sync_service: Arc<EncryptionSyncService>,
+    state: SharedObservable<State>,
+    encryption_sync_permit: Arc<AsyncMutex<EncryptionSyncPermit>>,
+    supervisor: Option<SyncTaskSupervisor>,
```

Comment:
Did we lose again all the doc comments here? :(

Reply:
Only one of those fields had docs, but indeed it got lost in a rebase for some reason: 20628615cdaeb6fd4764c2d81674b21293496919.